### DISCUSSION TITLE
Intermittent 406 errors

### DISCUSSION THREAD

**Discussion by GrimRob** [dotnetrdf/dotnetrdf#389] [Q&A]
I get intermittent rdf errors

```
var endpoint = new SparqlRemoteEndpoint(new Uri("http://dbpedia.org/sparql"), "http://dbpedia.org");
var query = "my query";
var rset = endpoint.QueryWithResultSet(query);
```

VDS.RDF.Query.RdfQueryException: 'A HTTP Error occurred while trying to make the SPARQL Query, see inner exception for details'

If I jump back in the debugger sometime it works

How do I get around it?

**Comment by GrimRob** [comment]
My query worked perfectly well in Virtuoso  so I just called the URL directly and didn't use dotnetrdf, then deserialised the response

**Comment by kal** [comment]
Hi @GrimRob,

That sounds like it could possibly be a problem with the SPARQL endpoint, rather than with the dotNetRDF library. But if you feel like digging a bit more, you could take a look at https://github.com/dotnetrdf/dotnetrdf/wiki/HowTo-Debug-HTTP-Communication for instructions on getting more detailed logging information from the HTTP connection. 

If you think this is maybe a problem with dotNetRDF, can you share the query that causes the problem so that I can do some testing from my location please?

Cheers

Kal

**Reply by GrimRob** [reply]
It's when you have a query that returns nothing

SELECT ?wikiPageRedirects WHERE
{
  <http://dbpedia.org/resource/Aymer,_Count_of_Angoulême> rdfs:label ?lbl .
  ?sub rdfs:label ?lbl .
  optional {?sub dbo:wikiPageRedirects ?wikiPageRedirects }
  FILTER (langMatches(lang(?lbl),"en"))
}

It's just something I catch and handle as "normal"

            catch (RdfQueryException ex)
            {
                if (ex?.InnerException.Message == "The remote server returned an error: (406) Not Acceptable.")
                {

Not really a problem but in an ideal world I'd rather it just was dealt with "in house"
